tables/TaQL/TableParseSortKey.cc

```diff
@@ -76,156 +76,144 @@ namespace casacore { //# NAMESPACE CASACORE - BEGIN
     switch (node_p.getColumnDataType()) {
     case TpBool:
       {
-        Array<Bool>* array = new Array<Bool>
-          (node_p.getColumnBool(rownrs));
-        arrPtr.reset (array);
+        auto array = std::make_shared<Array<Bool>>(node_p.getColumnBool(rownrs));
+        arrPtr = array;
         Bool* data = array->getStorage (deleteIt);
         if (deleteIt) {
           // A copy has been made, so replace the array.
-          arrPtr.reset (new Array<Bool> (array->shape(), data, TAKE_OVER));
+          arrPtr = std::make_shared<Array<Bool>>(array->shape(), data, TAKE_OVER);
         }
         sort.sortKey (data, TpBool, 0, order);
```

Comment:
Understanding this code is hard. It looks like the `getStorage` call forces using a contiguous memory chunk, so we can pass the `data` pointer to `sort.sortKey`.

If I'm correct, could you please change this code so it reflects that behavior. For example:
```
arrPtr = std::make_shared<Array<Bool>>(node_p.getColumnBool(rownrs));
if (!arrPtr->contiguousStorage()) arrPtr->unique(); // Make the array contiguous.
sort.sortKey (arrPtr->data(), TpBool, 0, order);
```
Please also update the other `case`s below.

Reply:
Avoid getStorage by testing if the array is contiguous.